site/app/models/gradeable/Gradeable.php

```diff
@@ -0,0 +1,469 @@
+ * @method string getSyllabusBucket();
+ * @method void setSyllabusBucket($bucket);
+ * @method Component[] getComponents();
+ * @method string getTaInstructions();
+ * @method void setTaInstructions($instructions);
+ * @method string getAutogradingConfigPath();
+ * @method bool isVcs();
+ * @method void setVcs($use_vcs);
+ * @method string getVcsSubdirectory();
+ * @method void setVcsSubdirectory($subdirectory);
+ * @method bool isTeamAssignment();
+ * @method int getTeamSizeMax();
+ * @method \DateTime getTeamLockDate();
+ * @method bool isTaGrading();
+ * @method void setTaGrading($use_ta_grading);
+ * @method bool isStudentView();
+ * @method void setStudentView($can_student_view);
+ * @method bool isStudentSubmit();
+ * @method void setStudentSubmit($can_student_submit);
+ * @method bool isStudentDownload();
+ * @method void setStudentDownload($can_student_download);
+ * @method bool isStudentDownloadAnyVersion();
+ * @method void setStudentDownloadAnyVersion($student_download_any_version);
+ * @method bool isPeerGrading();
+ * @method void setPeerGrading($use_peer_grading);
+ * @method int getPeerGradeSet();
+ * @method void setPeerGradeSet($grade_set)
+ * @method \DateTime getSubmissionOpenDate();
+ * @method \DateTime getSubmissionDueDate();
+ * @method int getLateDays();
+ * @method bool isLateSubmissionAllowed();
+ * @method void setLateSubmissionAllowed($allow_late_submission)
+ * @method float getPrecision();
+ * @method void setPrecision($grading_precision);
+ */
+class Gradeable extends AbstractModel
+{
+    /* Properties for all types of gradeables */
+
+    /** @property @var string The course-wide unique gradeable id */
+    protected $id = "";
+    /** @property @var string The gradeable's title */
+    protected $title = "";
+    /** @property @var string The instructions url to give to students */
+    protected $instructions_url = "";
+    /** @property @var int The type of gradeable */
+    protected $type = GradeableType::ELECTRONIC_FILE;
+    /** @property @var bool If the gradeable should be graded per registration section (true) or rotating sections(false) */
+    protected $grade_by_registration = true;
+    /** @property @var int The minimum user group that can grade this gradeable (1=instructor) */
+    protected $min_grading_group = 1;
+    /** @property @var string The syllabus classification of this gradeable */
+    protected $syllabus_bucket = "homework";
+
+    /* Properties exclusive to numeric-text/checkpoint gradeables */
+
+    /** @property @var string The overall ta instructions for grading (numeric-text/checkpoint only) */
+    protected $ta_instructions = "";
+
+    /* Properties exclusive to electronic gradeables */
+
+    /** @property @var string The location of the autograding configuration file */
+    protected $autograding_config_path = "";
+    /** @property @var string[] The object that contains the autograding config data */
+    private $autograding_config = null;
```

Comment:
Any reason this is private?

Reply:
Originally I wasn't planning on serializing it with the rest of the gradeable, but it probably makes sense to